core/domain/exp_services.py

```diff
@@ -684,6 +731,7 @@ def apply_change_list(exploration_id, change_list):
         Exception: Any entries in the changelist are invalid.
     """
     exploration = get_exploration_by_id(exploration_id)
+    image_ids_before_change = get_images_ids_of_exploration(exploration)
```

Comment:
Why are we doing this?

Reply:
``` Deleted image_ids = set(image_ids_before_change) - set(image_ids_after_change) ```

So, ``` image_ids_before_change ``` is the image ids that are present before doing editing state.

Reply:
no more implementation by the above method.
done